### PR TITLE
Move  "Configuring external authentication" to Installing Server

### PR DESCRIPTION
**Pull request by fiyyyu**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.5/Katello 4.7 (planned Satellite 6.13)
* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5 (Satellite 6.12)
* [ ] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3 (Satellite 6.11, orcharhino 6.1+)
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/common/modules/con_user-groups.adoc [hunk 1]
```diff
@@ -3,6 +3,6 @@
 
 With {Project}, you can assign permissions to groups of users.
 You can also create user groups as collections of other user groups.
-If using an external authentication source, you can map {Project} user groups to external user groups as described in xref:Configuring_External_User_Groups_{context}[].
+If using an external authentication source, you can map {Project} user groups to external user groups as described in link:{InstallingServerDocURL}Configuring_External_User_Groups_{project-context}[Configuring external user groups].
```
Comment (maximiliankolb) [inline guides/common/modules/con_user-groups.adoc:6]:
```suggestion
If using an external authentication source, you can map {Project} user groups to external user groups as described in {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}[Configuring external user groups] in _{InstallingServerDocTitle}_.
```

We always omit the `link:` prefix. Frr a list of titles, see https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-titles.adoc

File: guides/common/modules/proc_configuring-freeipa-authentication-on-server.adoc [hunk 2]
```diff
@@ -47,7 +47,7 @@ ifdef::satellite[]
 [WARNING]
 ====
 This command might restart {Project} services during the installation of the package.
-For more information about installing and updating packages on {Project}, see xref:Managing_Packages_on_the_Base_Operating_System_{context}[].
+For more information about installing and updating packages on {Project}, see link:{AdministeringDocURL}Managing_Packages_on_the_Base_Operating_System_{project-context}[Managing Packages on the Base Operating System of Satellite Server or Capsule Server].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-freeipa-authentication-on-server.adoc:50]: Please do not add downstream terminology as per our conventions.

**Review by maximiliankolb — COMMENTED** [review]
Hi @fiyyyu 

The move per se looks good to me. Please ensure to remove the `link:` prefix and add the title of the guide in italics.

File: guides/common/modules/con_user-groups.adoc (hunk 1 above)
Comment (fiyyyu) [reply guides/common/modules/con_user-groups.adoc]: thank you, I have removed the "link:" parts from all references. I have also added names of titles in italics

**Comment by fiyyyu** [comment]
Requested changes made in https://github.com/theforeman/foreman-documentation/commit/5a5072fb760e67913dc81f8b72348f7c6917231f

File: guides/common/modules/con_user-groups.adoc [hunk 3]
```diff
@@ -3,6 +3,6 @@
 
 With {Project}, you can assign permissions to groups of users.
 You can also create user groups as collections of other user groups.
-If using an external authentication source, you can map {Project} user groups to external user groups as described in xref:Configuring_External_User_Groups_{context}[].
+If using an external authentication source, you can map {Project} user groups to external user groups as described in {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}[Configuring external user groups] in _{InstallingServerDocTitle}_.
```
Comment (maximiliankolb) [inline guides/common/modules/con_user-groups.adoc:6]:
```suggestion
If using an external authentication source, you can map {Project} user groups to external user groups as described in {InstallingServerDocURL}Configuring_External_User_Groups_admin[Configuring external user groups] in _{InstallingServerDocTitle}_.
```

IMHO, xrefs to another document always require the hardcoded context from the guide itself. See line 3: https://raw.githubusercontent.com/theforeman/foreman-documentation/master/guides/doc-Administering_Project/master.adoc

Comment (maximiliankolb) [reply guides/common/modules/con_user-groups.adoc:6]: If a specific section is only part of one build target (for example Satellite), then yes, you'll need `ifdef::satellite[]` for the xref in any other file.

I've looked at the wrong guide :see_no_evil: 
You're absolutely right, it should be `{project-context}`: https://github.com/theforeman/foreman-documentation/blame/master/guides/doc-Installing_Server/master.adoc#L4

File: guides/common/modules/con_user-groups.adoc [hunk 4]
```diff
@@ -3,6 +3,13 @@
...
 You can also create user groups as collections of other user groups.
-If using an external authentication source, you can map {Project} user groups to external user groups as described in xref:Configuring_External_User_Groups_{context}[].
+
+ifdef::satellite[]
+If using an external authentication source, you can map {Project} user groups to external user groups as described in {InstallingServerDocURL}Configuring_External_User_Groups_satellite[Configuring external user groups] in _{InstallingServerDocTitle}_.
+endif::[]
+
+ifndef::satellite[]
+If using an external authentication source, you can map {Project} user groups to external user groups as described in {InstallingServerDocURL}Configuring_External_User_Groups_foreman[Configuring external user groups].
+endif::[]
```
Comment (maximiliankolb) [inline guides/common/modules/con_user-groups.adoc:13]:
```suggestion
If using an external authentication source, you can map {Project} user groups to external user groups as described in {InstallingServerDocURL}Configuring_External_User_Groups_satellite[Configuring external user groups] in _{InstallingServerDocTitle}_.
```

**Comment by maximiliankolb** [comment]
This section is currently part of the "Administering Project" guide (https://docs.theforeman.org/nightly/Administering_Project/index-katello.html#Configuring_External_Authentication_admin). Moving it generally means that you move it for all build targets. If you want this for Satellite, and I say "It's okay for orcharhino", and I also agree that it makes sense for the community, then you can make the move from one `master.adoc` to another.

All xrefs should then use full links using `...DocURL` and `...DocTitle`. As far as I can tell, there is no need to differentia between Satellite and non-Satellite.

I think the links should use `{project-context}` which evalutes as follows:

````
$ git grep ":project-context:" 
guides/common/attributes-base.adoc::project-context: foreman
guides/common/attributes-orcharhino.adoc::project-context: orcharhino
guides/common/attributes-satellite.adoc::project-context: satellite
````

Since this attribute is never overwritten or changed in any guide, you should be pretty safe to use it as is.

I made a suggestion which should basically work for all xrefs.

File: guides/common/modules/con_user-groups.adoc (hunk 4 above)
Comment (fiyyyu) [reply guides/common/modules/con_user-groups.adoc]: Thanks very much for the additional info. What confused me was the fact that the {project-context} translates into "admin" in the "Administering Red Hat Satellite" guide: see e.g. https://access.redhat.com/documentation/en-us/red_hat_satellite/6.12/html-single/administering_red_hat_satellite/index#Starting_and_Stopping_Server_admin 

At any rate, I have implemented your suggestions (haven't accepted the ^^ commit as this would lead to a merge conflict):
- I have replaced the hard values with {project-context} in the links
- I have removed the ifdef/ifndef statements and only left the version that uses  ...DocURL and ...DocTitle
The title builds when I run "make BUILD=satellite" and "make BUILD=katello".

I hope the fork is ready to be merged to master now. Please let me know if not.

File: guides/common/modules/con_user-groups.adoc [hunk 5]
```diff
@@ -3,6 +3,7 @@
 
 With {Project}, you can assign permissions to groups of users.
 You can also create user groups as collections of other user groups.
-If using an external authentication source, you can map {Project} user groups to external user groups as described in xref:Configuring_External_User_Groups_{context}[].
+
```
Comment (maximiliankolb) [inline guides/common/modules/con_user-groups.adoc:6]:
```suggestion
```

File: guides/common/modules/proc_configuring-external-user-groups.adoc [hunk 6]
```diff
@@ -16,8 +16,7 @@ For more information see xref:Using_LDAP_{context}[].
 +
 When using external user groups from an LDAP source, you cannot use the `$login` variable as a substitute for the account user name.
 You must use either an anonymous or dedicated service user.
-* If you use a {FreeIPA} or AD server, configure {Project} to use {FreeIPA} or AD authentication.
-For more information, see xref:Configuring_External_Authentication_{context}[].
+* If you use a {FreeIPA} or AD server, configure {Project} to use {FreeIPA} or AD authentication. For more information, see {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring external authentication] in _{InstallingServerDocTitle}_.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-external-user-groups.adoc:19]:
```suggestion
* If you use a {FreeIPA} or AD server, configure {Project} to use {FreeIPA} or AD authentication.
For more information, see {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring external authentication] in _{InstallingServerDocTitle}_.
```

**Comment by fiyyyu** [comment]
I hope I have done the right thing by accepting your newest suggestions.
Strangely, you seem to have suggested a change (xref->external link) that I had already made locally (and pushed remotely) before. 

It's the first time I have worked on an upstream project like this. The workflow on the RH Identity Management team, where I am based, is very different -> thanks for your patience.

**Review by maximiliankolb — APPROVED** [review]
Thank you @fiyyyu for your first contribution to Foreman docs! Your PR LGTM!

But since it's a rather big structural change, I'm handing over to @adamlazik1 for an additional review.

File: guides/common/modules/con_user-groups.adoc (hunk 3 above)
Comment (adamlazik1) [inline guides/common/modules/con_user-groups.adoc:6]:
```suggestion
If using an external authentication source, you can map {Project} user groups to external user groups as described in {InstallingServerDocURL}Configuring_External_User_Groups_{project-context}[Configuring External User Groups] in _{InstallingServerDocTitle}_.
```

File: guides/common/modules/proc_configuring-external-user-groups.adoc [hunk 7]
```diff
@@ -17,7 +17,7 @@ For more information see xref:Using_LDAP_{context}[].
 When using external user groups from an LDAP source, you cannot use the `$login` variable as a substitute for the account user name.
 You must use either an anonymous or dedicated service user.
 * If you use a {FreeIPA} or AD server, configure {Project} to use {FreeIPA} or AD authentication.
-For more information, see xref:Configuring_External_Authentication_{context}[].
+For more information, see {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring external authentication] in _{InstallingServerDocTitle}_.
```
Comment (adamlazik1) [inline guides/common/modules/proc_configuring-external-user-groups.adoc:20]:
```suggestion
For more information, see {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring External Authentication] in _{InstallingServerDocTitle}_.
```

File: guides/common/modules/proc_configuring-project-to-use-ldap.adoc [hunk 8]
```diff
@@ -31,8 +31,8 @@ Selected organizations are assigned to users created from the LDAP authenticatio
 . Click *Submit*.
 . Configure new accounts for LDAP users:
 +
-* If you did not select *Automatically Create Accounts In {Project}* checkbox, see xref:Creating_a_User_{context}[] to create user accounts manually.
+* If you did not select *Automatically Create Accounts In {Project}* checkbox, see {AdministeringDocURL}Creating_a_User_{project-context}[Creating a user] in _{AdministeringDocTitle}_ to create user accounts manually.
```
Comment (adamlazik1) [inline guides/common/modules/proc_configuring-project-to-use-ldap.adoc:34]:
```suggestion
* If you did not select *Automatically Create Accounts In {Project}* checkbox, see {AdministeringDocURL}Creating_a_User_{project-context}[Creating a User] in _{AdministeringDocTitle}_ to create user accounts manually.
```

File: guides/common/modules/proc_creating-a-user.adoc [hunk 9]
```diff
@@ -18,7 +18,7 @@ By default, {ProjectServer} uses the language and timezone settings of the user
 . Set a password for the user:
 .. From the *Authorized by* list, select the source by which the user is authenticated.
 ** *INTERNAL*: to enable the user to be managed inside {ProjectServer}.
-** *EXTERNAL*: to configure external authentication as described in xref:Configuring_External_Authentication_{context}[].
+** *EXTERNAL*: to configure external authentication as described in {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring external authentication] in _{InstallingServerDocTitle}_.
```
Comment (adamlazik1) [inline guides/common/modules/proc_creating-a-user.adoc:21]:
```suggestion
** *EXTERNAL*: to configure external authentication as described in {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring External Authentication] in _{InstallingServerDocTitle}_.
```

File: guides/common/modules/proc_renaming-server.adoc [hunk 10]
```diff
@@ -6,7 +6,7 @@ This procedure ensures that you update all references to the new host name.
 
 If you use external authentication, you must reconfigure {ProjectServer} for external authentication after you run the `{project-change-hostname}` script.
 The `{project-change-hostname}` script breaks external authentication for {ProjectServer}.
-For more information about configuring external authentication, see xref:Configuring_External_Authentication_{context}[].
+For more information about configuring external authentication, see {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring external authentication] in _{InstallingServerDocTitle}_.
```
Comment (adamlazik1) [inline guides/common/modules/proc_renaming-server.adoc:9]:
```suggestion
For more information about configuring external authentication, see {InstallingServerDocURL}Configuring_External_Authentication_{project-context}[Configuring External Authentication] in _{InstallingServerDocTitle}_.
```

**Review by adamlazik1 — COMMENTED** [review]
Looks good to me! The only note I have is that we use camel case for titles.

**Comment by fiyyyu** [comment]
Thanks, Adam. I have accepted your suggestions by committing the changes. 
Is there anything else that must be done before this pull request is merged?

**Review by adamlazik1 — APPROVED** [review]
LGTM. Please indicate the versions you want this to get merged into in the PR message.

File: guides/common/modules/proc_configuring-project-to-use-ldap.adoc [hunk 11]
```diff
@@ -31,8 +31,8 @@ Selected organizations are assigned to users created from the LDAP authenticatio
 . Click *Submit*.
 . Configure new accounts for LDAP users:
 +
-* If you did not select *Automatically Create Accounts In {Project}* checkbox, see xref:Creating_a_User_{context}[] to create user accounts manually.
+* If you did not select *Automatically Create Accounts In {Project}* checkbox, see {AdministeringDocURL}Creating_a_User_{project-context}[Creating a User] in _{AdministeringDocTitle}_ to create user accounts manually.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-project-to-use-ldap.adoc:34]:
```suggestion
* If you did not select *Automatically Create Accounts In {Project}* checkbox, see {AdministeringDocURL}Creating_a_User_admin[Creating a User] in _{AdministeringDocTitle}_ to create user accounts manually.
```

See https://github.com/theforeman/foreman-documentation/blame/master/guides/doc-Administering_Project/master.adoc#L3 or run `git grep "{AdministeringDocURL}"` to see some examples.

File: guides/common/modules/proc_configuring-project-to-use-ldap.adoc [hunk 12]
```diff
@@ -31,8 +31,8 @@ Selected organizations are assigned to users created from the LDAP authenticatio
...
 +
 * If you selected the *Automatically Create Accounts In {Project}* checkbox, LDAP users can now log in to {Project} using their LDAP accounts and passwords.
 After they log in for the first time, the {Project} administrator has to assign roles to them manually.
-See xref:Assigning_Roles_to_a_User_{context}[] to assign user accounts appropriate roles in {Project}.
+See {AdministeringDocURL}Assigning_Roles_to_a_User_{project-context}[Assigning Roles to a User] in _{AdministeringDocTitle}_ to assign user accounts appropriate roles in {Project}.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-project-to-use-ldap.adoc:38]:
```suggestion
For more information on assigning user accounts appropriate roles in {Project}, see {AdministeringDocURL}Assigning_Roles_to_a_User_admin[Assigning Roles to a User] in _{AdministeringDocTitle}_.
```

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Please update both links and select which Foreman version you want your change to have applied to. Merging to master means Foreman nightly, so it'll only be part of the next Foreman release.

File: guides/common/modules/proc_configuring-project-to-use-ldap.adoc (hunk 11 above)
Comment (fiyyyu) [reply guides/common/modules/proc_configuring-project-to-use-ldap.adoc:34]: Well I am confused. In your comment above, posted 3 weeks ago, you said the following: "I think the links should use {project-context} ...". 
For this reason, I replaced all "admin" occurrences with {project-context}. 
Now you seem to be saying that I should do the reverse operation, i.e. replace {project-context} with "admin".
Can you please explain if there is something I am getting wrong?

File: guides/common/modules/proc_configuring-project-to-use-ldap.adoc (hunk 12 above)
Comment (fiyyyu) [reply guides/common/modules/proc_configuring-project-to-use-ldap.adoc:38]: Same comment as above:
Well I am confused. In your comment above, posted 3 weeks ago, you said the following: "I think the links should use {project-context} ...". 
For this reason, I replaced all "admin" occurrences with {project-context}. 
Now you seem to be saying that I should do the reverse operation, i.e. replace {project-context} with "admin".
Can you please explain if there is something I am getting wrong?

**Comment by fiyyyu** [comment]
Hi all,
I have checked the "Foreman 3.5/Katello 4.7 (planned Satellite 6.13)" and "Foreman 3.3/Katello 4.5 (Satellite 6.12)" check boxes above to indicate in which versions I would like to see the changes to occur. 
This seems to me the most reasonable application of the changes I suggest but I am happy to check whatever you advise me to check. I am not familiar with these sort of "backporting" standards in Satellite docs.

File: guides/common/modules/proc_configuring-project-to-use-ldap.adoc (hunk 11 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_configuring-project-to-use-ldap.adoc:34]: I also got confused by the other example which in fact uses `{project-context}` as context instead of a hardcoded, different value for each guide. Examples:

* [`master.adoc` in Tuning Foreman](https://github.com/theforeman/foreman-documentation/blame/master/guides/doc-Tuning_Performance/master.adoc#L3) uses `performance-tuning
` which is then "appended" to all links: https://docs.theforeman.org/nightly/Tuning_Performance/index-katello.html#Apache_HTTPD_Performance_Tuning_performance-tuning
* [`master.adoc` in Managing Content](https://github.com/theforeman/foreman-documentation/blame/master/guides/doc-Managing_Content/master.adoc#L3) uses `content-management` which results in links such as https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Creating_an_Activation_Key_to_Consume_the_Content_View_content-management

The "Administering Project" guide uses `admin`: https://github.com/theforeman/foreman-documentation/blame/master/guides/doc-Administering_Project/master.adoc#L3

IMHO there is only few guides that use `{project-context}` as context:

````
$ git grep "^:context:" **/master.adoc
guides/doc-Administering_Project/master.adoc::context: admin
guides/doc-Administering_with_Ansible/master.adoc::context: ansible-admin
guides/doc-Configuring_Load_Balancer/master.adoc::context: load-balancing
guides/doc-Deploying_Hosts_AppCentric/master.adoc::context: application_centric_deployment
guides/doc-Deploying_Project_on_AWS/master.adoc::context: deploying-on-aws
guides/doc-Installing_Proxy/master.adoc::context: {smart-proxy-context}
guides/doc-Installing_Server/master.adoc::context: {project-context}
guides/doc-Installing_Server_Disconnected/master.adoc::context: satellite
guides/doc-Managing_Configurations_Ansible/master.adoc::context: ansible
guides/doc-Managing_Configurations_Puppet/master.adoc::context: managing-configurations-puppet
guides/doc-Managing_Content/master.adoc::context: content-management
guides/doc-Managing_Hosts/master.adoc::context: managing-hosts
guides/doc-Managing_Organizations_and_Locations/master.adoc::context: managing-organizations-locations
guides/doc-Planning_for_Project/master.adoc::context: planning
guides/doc-Provisioning_Hosts/master.adoc::context: provisioning
guides/doc-Quickstart/master.adoc::context: {project-context}
guides/doc-Tuning_Performance/master.adoc::context: performance-tuning
guides/doc-Upgrading_and_Updating/master.adoc::context: upgrade-guide
````

For orcharhino, the downstream product of my employer, I don't use installing Server/Proxy and Quickstart guides, so I had less experience with them.

Long story short: if you link to the "Administering Project" guide using `{AdministeringDocURL}`, set the context to "admin".

Regarding cherry-picks: I've also ticked the box for 3.4, because this makes cherry-picked possible commits on top easier. Requesting cherry-picks to both 3.5 and 3.3 is almost always the case, unless we're documenting cutting edge things.

Comment (fiyyyu) [reply guides/common/modules/proc_configuring-project-to-use-ldap.adoc:34]: Thank you for the explanation. I will check and set the correct context this evening.

**Comment by fiyyyu** [comment]
I have now replaced {project-context} with admin in links to the doc-Adminstering_Project guide as discussed earlier today. For details see 
https://github.com/theforeman/foreman-documentation/pull/1878/commits/08b55feb7961c3d726527668cb8e060712913bfa

Hopefully now all is ready for the PR to be approved. Let me know if not.

**Review by maximiliankolb — APPROVED** [review]
LGTM! Thanks @fiyyyu for your first contribution for foreman-documentation. From now on, your GHA pipelines will start automatically.

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: :heavy_check_mark: 
   149b61cc5..bad888402  3.5 -> 3.5
   171e1c36c..2b349bd93  3.4 -> 3.4
   644ee74a7..93af0231e  3.3 -> 3.3

FYI: I've squashed your commits.

**Comment by fiyyyu** [comment]
Thanks very much, @maximiliankolb! I am now closing the internal RH ticket, too. 
It looks like I will be opening a new pull request soon, because of https://bugzilla.redhat.com/show_bug.cgi?id=2157599
This new update is a trivial one though.
